src/server/JasmineGraphInstanceService.cpp

```diff
@@ -556,3 +636,339 @@ void writeCatalogRecord(string record) {
+
+                    unordered_set<long> aggregatedEndVidSet = aggregatedCentralStore[startVid];
+                    aggregatedEndVidSet.insert(endVidSet.begin(),endVidSet.end());
+                    aggregatedCentralStore[startVid] = aggregatedEndVidSet;
+                }
+            }
+        }
+
+    }
+    map<long, long> distributionHashMap = JasmineGraphInstanceService::getOutDegreeDistributionHashMap(aggregatedCentralStore);
+    long aggregatedTriangleCount = JasmineGraphInstanceService::countCentralStoreTriangles(aggregatedCentralStore,distributionHashMap);
+
+    return aggregatedTriangleCount;
+
+}
+
+map<long, long> JasmineGraphInstanceService::getOutDegreeDistributionHashMap(map<long, unordered_set<long>> graphMap) {
+    map<long, long> distributionHashMap;
+
+    for (map<long, unordered_set<long>>::iterator it = graphMap.begin(); it != graphMap.end(); ++it) {
+        long distribution = (it->second).size();
+        distributionHashMap.insert(std::make_pair(it->first, distribution));
+    }
+    return distributionHashMap;
+}
+
+
+long JasmineGraphInstanceService::countCentralStoreTriangles(map<long, unordered_set<long>> centralStore,
+                                                             map<long, long> distributionMap) {
+   std::map<long,long> degreeReverseLookupMap;
+   std::vector<std::set<long>> degreeVector;
+    std::map<long,std::set<long>> degreeMap;
+
+    long startVId;
+    long degree;
+    int maxDegree=0;
+
+    std::map<long,long>::iterator it;
+    std::map<long,long>::iterator degreeDistributionIterator;
+    std::map<long,long>::iterator centralDBDegreeDistributionIterator;
+
+
+    for (it = distributionMap.begin(); it != distributionMap.end();++it) {
+        degree = it->second;
+
+        if (degree > maxDegree) {
+            maxDegree=degree;
+        }
+
+        /*if (degreeVector.size() > degree) {
+            degreeSet = degreeVector[degree];
+            degreeSet.insert(startVId);
+        } else {
+            degreeVector.resize(degree+1);
+            degreeSet.insert(startVId);
+        }*/
+
+        //degreeVector.at(degree) = degreeSet;
+    }
+
+    for (int degreeIndex=0;degreeIndex<=maxDegree;degreeIndex++) {
+        std::set<long> degreeSet;
+        for (it = distributionMap.begin(); it != distributionMap.end();++it) {
+            startVId = it->first;
+            degree = it->second;
+
+            if (degreeIndex == degree) {
+                degreeSet.insert(startVId);
+            }
+
+        }
+        degreeMap[degreeIndex]=degreeSet;
+    }
+
+    long triangleCount = 0;
+    long varOne = 0;
+    long varTwo = 0;
+    long varThree = 0;
+    long fullCount = 0;
+
+    std::map<long, std::map<long, std::vector<long>>> triangleTree;
+    std::vector<long> degreeListVisited;
+
+    std::map<long,std::set<long>>::iterator iterator;
+
+    for (iterator = degreeMap.begin(); iterator != degreeMap.end();++iterator) {
```

Comment:
Lets move this triangle counting logic to query/algorithms package. Otherwise the logic will be duplicated